From 004216f8af3661621a23516eae63a421033a6c38 Mon Sep 17 00:00:00 2001
From: Raymond Tan <raymond.tan@intel.com>
Date: Mon, 20 Jul 2020 17:09:11 +0800
Subject: [PATCH] usb: dwc3: Allow Elkhart Lake to utilize DSM method for PM
 functionality

Similar to some other IA platforms, Elkhart Lake too depends on the
PMU register write to indicate D0/D3 state to power manage the controller.

Thus, we add the PCI_DEVICE_ID_INTEL_EHLLP to the list of devices
that shall execute ACPI _DSM method during D0/D3 sequence.

Signed-off-by: Raymond Tan <raymond.tan@intel.com>
---
 drivers/usb/dwc3/dwc3-pci.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/dwc3-pci.c b/drivers/usb/dwc3/dwc3-pci.c
index b673727..362c260 100644
--- a/drivers/usb/dwc3/dwc3-pci.c
+++ b/drivers/usb/dwc3/dwc3-pci.c
@@ -145,7 +145,8 @@ static int dwc3_pci_quirks(struct dwc3_pci *dwc)
 
 	if (pdev->vendor == PCI_VENDOR_ID_INTEL) {
 		if (pdev->device == PCI_DEVICE_ID_INTEL_BXT ||
-				pdev->device == PCI_DEVICE_ID_INTEL_BXT_M) {
+				pdev->device == PCI_DEVICE_ID_INTEL_BXT_M ||
+				pdev->device == PCI_DEVICE_ID_INTEL_EHLLP) {
 			guid_parse(PCI_INTEL_BXT_DSM_GUID, &dwc->guid);
 			dwc->has_dsm_for_pm = true;
 		}
-- 
2.7.4

